apps/emqx/src/emqx_channel.erl

```diff
@@ -782,6 +782,19 @@ maybe_update_expiry_interval(_Properties, Channel) -> Channel.
 
 -spec(handle_deliver(list(emqx_types:deliver()), channel())
       -> {ok, channel()} | {ok, replies(), channel()}).
+
+handle_deliver(Delivers, Channel = #channel{takeover = true,
```

Comment:
Just wondering, couldn't we explicitly match on `takeover := false` in other clauses 🤔 So that we do not depend on the order?

Reply:
This would be even safer. The second clause also does the same thing with `conn_state`. I think the reordering makes sense anyway,